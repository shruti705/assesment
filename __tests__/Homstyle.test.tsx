import styles from '../src/component/HomeScreen/HomeStyle';

jest.mock('react-native', () => {
    const ReactNative = jest.requireActual('react-native');
    return {
      ...ReactNative,
      StyleSheet: {
        create: jest.fn((styles) => styles),
      },
    };
  });
  
describe('StyleSheet', () => {
  it('should match the snapshot', () => {
    expect(styles).toMatchSnapshot();
  });

  it('should have correct container styles', () => {
    expect(styles.container).toEqual({
      flex: 1,
      backgroundColor: '#1a2330',
    });
  });

  it('should have correct subContainer styles', () => {
    expect(styles.subContainer).toEqual({
      marginTop : "10%",
      marginLeft : "7%",
    });
  });

  it('should have correct image styles', () => {
    expect(styles.image).toEqual({
      width: 166,
      resizeMode: "contain",
    });
  });

  it('should have correct textContainer styles', () => {
    expect(styles.textContainer).toEqual({
      marginTop: "20%",
      justifyContent: "center",
      alignItems: "center",
      paddingRight: "5%",
      paddingLeft: "5%",
    });
  });

  it('should have correct textSubHeader styles', () => {
    expect(styles.textSubHeader).toEqual({
      marginTop: "2%",
      fontSize: 18,
      fontWeight: "400",
      lineHeight: 20,
      color: "white",
    });
  });

  it('should have correct textInput styles', () => {
    expect(styles.textInput).toEqual({
      width: "100%",
      height: 56,
      marginTop : "4%",
      paddingLeft: "5%",
      color: "black",
      borderWidth: 2,
      borderRadius: 5,
      backgroundColor: "white",
      textAlign: "left",
      borderColor: "white",
    });
  });
});
