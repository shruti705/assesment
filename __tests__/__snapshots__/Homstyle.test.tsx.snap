// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleSheet should match the snapshot 1`] = `
{
  "container": {
    "backgroundColor": "#1a2330",
    "flex": 1,
  },
  "image": {
    "resizeMode": "contain",
    "width": 166,
  },
  "modal": {
    "alignItems": "center",
    "justifyContent": "center",
  },
  "modalButton": {
    "backgroundColor": "#1a2330",
    "borderRadius": 5,
    "padding": "5%",
  },
  "modalButtonText": {
    "color": "white",
    "fontSize": 16,
  },
  "modalContent": {
    "alignItems": "center",
    "backgroundColor": "white",
    "borderRadius": 10,
    "padding": "5%",
  },
  "modalText": {
    "color": "black",
    "fontSize": 18,
    "marginBottom": "5%",
  },
  "modalTextTitle": {
    "color": "black",
    "fontSize": 18,
    "marginBottom": "5%",
  },
  "subContainer": {
    "marginLeft": "7%",
    "marginTop": "10%",
  },
  "textContainer": {
    "alignItems": "center",
    "justifyContent": "center",
    "marginTop": "20%",
    "paddingLeft": "5%",
    "paddingRight": "5%",
  },
  "textInput": {
    "backgroundColor": "white",
    "borderColor": "white",
    "borderRadius": 5,
    "borderWidth": 2,
    "color": "black",
    "height": 56,
    "marginTop": "4%",
    "paddingLeft": "5%",
    "textAlign": "left",
    "width": "100%",
  },
  "textSubHeader": {
    "color": "white",
    "fontSize": 18,
    "fontWeight": "400",
    "lineHeight": 20,
    "marginTop": "2%",
  },
  "title_welcome": {
    "color": "white",
    "fontSize": 25,
    "fontWeight": "700",
    "marginLeft": "3%",
    "marginTop": "5%",
  },
}
`;
